modules/nextflow/src/main/groovy/nextflow/executor/LsfExecutor.groovy

```diff
@@ -83,63 +93,41 @@ class LsfExecutor extends AbstractGridExecutor {
+                def mem = task.config.getMemory()
+                // LSF mem limit can be both per-process and per-job
+                if( task.config.cpus > 1 && !perJobMemLimit ) {
+                    long bytes = mem.toBytes().intdiv(task.config.cpus as int)
+                    result << '-M' << String.valueOf(MemoryUnit.of(bytes).toUnit(memoryUnitLSF))
                 }
-                else { 
-                    new throw Exception ("LSF_UNIT_FOR_LIMITS not in KB, MB, or GB. Check variable $LSF_ENVDIR/lsf.conf")
+                else {
+                    result << '-M' << String.valueOf(mem.toUnit(memoryUnitLSF))
                 }
-            }
-            else {
-                if (memory_lsf_unit_for_limits.toUpperCase() in ['KB']){
+                result << '-R' << "select[mem>=${mem.toUnit(memoryUnitLSF)}] rusage[mem=${mem.toUnit(memoryUnitLSF)}]".toString()
+            } else{
+                // use default KB
+                def mem = task.config.getMemory()
+                // LSF mem limit can be both per-process and per-job
+                if( task.config.cpus > 1 && !perJobMemLimit ) {
+                    long bytes = mem.toBytes().intdiv(task.config.cpus as int)
                     result << '-M' << String.valueOf(MemoryUnit.of(bytes).toKilo())
-                    result << '-R' << "select[mem>=${mem.toKilo()}] rusage[mem=${mem.toKilo()}]".toString()
-                }
-                else if (memory_lsf_unit_for_limits.toUpperCase() in ['MB']){
-                    result << '-M' << String.valueOf(MemoryUnit.of(bytes).toMega())
-                    result << '-R' << "select[mem>=${mem.toMega()}] rusage[mem=${mem.toMega()}]".toString()
                 }
-                else if (memory_lsf_unit_for_limits.toUpperCase() in ['GB']){
-                    result << '-M' << String.valueOf(MemoryUnit.of(bytes).toGiga())
-                    result << '-R' << "select[mem>=${mem.toGiga()}] rusage[mem=${mem.toGiga()}]".toString()
-                }
-                else { 
-                    new throw Exception ("LSF_UNIT_FOR_LIMITS not in KB, MB, or GB. Check variable $LSF_ENVDIR/lsf.conf")
+                else {
+                    result << '-M' << String.valueOf(mem.toKilo())
                 }
+                result << '-R' << "select[mem>=${mem.toKilo()}] rusage[mem=${mem.toKilo()}]".toString()
             }
         }
 
         def disk = task.config.getDisk()
         if( disk ) {
-            if (memory_lsf_unit_for_limits.toUpperCase() in ['KB']){
-                result << '-R' << "select[tmp>=$disk.kilo] rusage[tmp=$disk.kilo]".toString()
-            }
-            else if (memory_lsf_unit_for_limits.toUpperCase() in ['MB']){
-                result << '-R' << "select[tmp>=$disk.mega] rusage[tmp=$disk.mega]".toString()
+            if ( memoryUnitLSF ){ 
+                result << '-R' << "select[tmp>=$disk.toUnit(memoryUnitLSF)] rusage[tmp=$disk.toUnit(memoryUnitLSF)]".toString()
```

Comment:
I'm not crazy about this solution. 

Some of my unease is it's not clear to me where `.kilo`, `.mega`, etc. are defined in `MemoryUnits.groovy`, though I see them in the tests: https://github.com/nextflow-io/nextflow/blob/master/modules/nf-commons/src/test/nextflow/util/MemoryUnitTest.groovy#L84-L85

Reply:
In groovy `.kilo`, `.mega` are shortcuts for `.getKilo()`, `.getMega()`, etc. In groovy any `.getXxx()` method can be accessed as an attribute `.xxx`.

Reply:
> In groovy `.kilo`, `.mega` are shortcuts for `.getKilo()`, `.getMega()`, etc. In groovy any `.getXxx()` method can be accessed as an attribute `.xxx`.

Ah, fascinating. This makes sense, and solves things nicely. I was somewhat confused, thanks